src/main/java/com/es/phoneshop/model/cart/service/DefaultCartService.java

```diff
@@ -61,21 +56,26 @@ public void add(HttpSession currentSession, Cart cart, Long productId, int quant
         }
     }
 
-    private int getAvailableProductQuantityToAddToCart(CartItem cartItem, Product product) {
-        if (cartItem == null) {
-            return product.getStock();
-        }
-        int currentProductCartQuantity = cartItem.getQuantity();
-        int productStock = cartItem.getProduct().getStock();
-        return productStock - currentProductCartQuantity;
+    private int getAvailableProductQuantity(Optional<CartItem> cartItem, Product product) {
```

Comment:
i don't recommend to use Optional as method argument:
https://stackoverflow.com/questions/31922866/why-should-java-8s-optional-not-be-used-in-arguments